doc_src/prompt.rst

```diff
@@ -0,0 +1,189 @@
+Writing your own prompt
+=======================
+
+Fish ships a number of prompts that you can view with the :doc:`fish_config <cmds/fish_config>` command, and many users have shared their prompts online.
+
+However, if you are anything like the author of this document, you sooner or later want to write your own or at least adjust the prompt you have. This is a good way to get used to fish's :doc:`scripting language <language>`.
+
+Unlike other shells, fish's prompt is built by running a function - :doc:`fish_prompt <cmds/fish_prompt>`. Or, more specifically, three functions:
+
+- :doc:`fish_prompt <cmds/fish_prompt>`
+- :doc:`fish_right_prompt <cmds/fish_right_prompt>` - which is shown on the right side of the terminal.
+- :doc:`fish_mode_prompt <cmds/fish_mode_prompt>` - which it runs if :ref:`vi-mode <vi-mode>` is used.
+
+These functions are run, and whatever they print is displayed as the prompt (minus one trailing newline).
+
+Here, we will just be writing a simple fish_prompt.
+
+.. note::
+   This document uses formatting to show what a prompt would look like. If you are viewing this in the man page,
+   you probably want to switch to looking at the web version instead. Run ``help custom-prompt`` to view it.
+
+Our first prompt
+----------------
+
+Let's look at a very simple example::
+
+  function fish_prompt
+      echo $PWD '>'
+  end
+
+This just prints the current working directory (:envvar:`PWD`) and a ``>`` symbol to show where the prompt ends. The ``>`` is :ref:`quoted <quotes>` because otherwise it would signify a :ref:`redirection <redirects>`.
+
+Because we've use :doc:`echo <cmds/echo>`, it adds spaces between the two so it ends up looking like (assuming ``_`` is your cursor):
+
+.. role:: white
+.. parsed-literal::
+    :class: highlight
+
+    :white:`/home/tutorial >`\ _
+
+Formatting
+----------
+
+If you don't want the space, there are other ways to print this:
+
+- ``echo -s``, which tells echo to not print a space.
+- We could construct one string: ``echo "$PWD>"``
+- :doc:`printf <cmds/printf>`, which allows us to give a format string to format our... string
+- :doc:`string join <cmds/string-join>`, if we want to have one thing appear between all our components.
```

Comment:
I'd rearrange this to be "If you don't want the space, you can use string join, like this:

Other methods include..." (or drop them entirely)

Reply:
You're right, this is a bit long. I mentioned `echo -s` and `printf`, but just by reference.